sdk/core/core-amqp/src/errors.ts

```diff
@@ -622,6 +639,8 @@ export function translate(err: AmqpError | Error): MessagingError {
     error = new MessagingError("Websocket connection failed.");
     error.name = ConditionErrorNameMapper[ErrorNameConditionMapper.ServiceCommunicationError];
     error.retryable = false;
+  } else if (customError) {
+    return customError;
```

Comment:
Since we are creating the custom error up front and using it only after a series of if/elses, the current model slightly hurts code readability and the ease of understanding. 

I would suggest the below
```typescript
if (isAmqpError(err)) { // do the needful and return the error }
if (isSystemError(err)) { // do the needful and return the error }
if (isBrowserWebsocketError(err)) { // do the needful and return the error }
const customError = getCustomError(err)
if (customError) {
   return customError
}
// The generic error handling here
```

Reply:
Please take another look at the suggested code change above.
Apart from moving the call to `getCustomError()`, the suggestion was to add return the error from each of the `if` blocks and avoiding the `else` blocks.

With https://github.com/Azure/azure-sdk-for-js/pull/4446/commits/f5f04287a0553874f889c1e324bad716f037ab71, you have moved the call to `getCustomError` and removed the `else` but are not returning the error object from the previous `if` blocks.

Due to this regardless of all the processing that is done by `if (isAmqpError(err))`, `if (isSystemError(err))` and `if (isBrowserWebsocketError(err)) ` all the non custom errors are getting converted to a generic error. This is also the reason why the tests are failing